Build error with older gcc

### Is there an existing issue for this?

- [x] I have searched the existing issues

### Midnight Commander version and build configuration

```shell
n/a
```

### Operating system

```shell
n/a
```

### Is this issue reproducible using the latest version of Midnight Commander?

- [ ] I confirm the issue is still reproducible with the latest version of Midnight Commander

### How to reproduce

when building with gcc 4.3.2, I get the compilation error:
```
  CC       editcmd.lo
editcmd.c: In function 'editcmd_check_and_create_user_syntax_directory':
editcmd.c:796: error: 'errno' undeclared (first use in this function)
editcmd.c:796: error: (Each undeclared identifier is reported only once
editcmd.c:796: error: for each function it appears in.)
editcmd.c:796: error: 'EEXIST' undeclared (first use in this function)
make[6]: *** [Makefile:527: editcmd.lo] Error 1
```

### Expected behavior

successful build

### Actual behavior

The build of `editcmd.c` fails because it is missing the include of <errno.h> (with the definition of the error codes like `EEXIST`)


### Additional context

The error was introduced by the commit 503c0af

It can simply be fixed with 
```
#include <errno.h>
```

Newer compilers do not have this problem, but it would be great if mc builds are fixed for older ones (we build mc as part of synocli-file tools for Synology NAS Devices at https://github.com/SynoCommunity/spksrc)

--- a/src/editor/editcmd.c
+++ b/src/editor/editcmd.c
@@ -43,6 +43,7 @@
 #include <string.h>
 #include <sys/stat.h>
 #include <stdlib.h>
+#include <errno.h>
 
 #include "lib/global.h"
 #include "lib/tty/tty.h"

